platform/linux-generic/include/odp/api/plat/atomic_inlines.h

```diff
@@ -333,6 +340,19 @@ _ODP_INLINE void odp_atomic_min_u64(odp_atomic_u64_t *atom, uint64_t new_val)
 	}
 }
 
+_ODP_INLINE uint64_t odp_atomic_fetch_max_u64(odp_atomic_u64_t *atom, uint64_t new_val)
+{
+	uint64_t old_val;
+
+	old_val = odp_atomic_load_u64(atom);
+
+	while (new_val > old_val) {
+		if (odp_atomic_cas_u64(atom, &old_val, new_val))
```

Comment:
Now I realized that this odp_atomic_cas_u64 here is just the spinlock version a little bit above. And so is the atomic load too. It would be much more efficient to implement this using the ATOMIC_OP macro so that the spinlock is taken once instead of two or more times.

Reply:
Changed in V5.